src/main/webapp/webapptests/spec/SpecManageSessions.js

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe("Manage Sessions", function() {
+
+    describe("when the student requests to see a their scheduled sessions", function() {
+        var mockWindow = {location: {href: "manage-sessions.html?studentEmail=test%40gmail.com", search: "?studentEmail=test%40gmail.com"}};
+
+        it("should trigger the fetch function", function() {
+            spyOn(window, "onload").and.callFake(function() {
+                readTutorID(queryString, window);
+            });
+            spyOn(window, 'fetch').and.callThrough();
+            getTutorSessionsManage(mockWindow);
+            expect(window.fetch).toHaveBeenCalledWith('/confirmation?studentEmail=undefined', {method: 'GET'});
```

Comment:
Why is the studentEmail undefined here?

Reply:
It's undefined because that is dependent on the that function that reads the studentEmail component into the queryString. That function is not being triggered here so it is not reading the url at all, that's why the studentEmail ends up being undefined. I am testing that it reads the studentEmail in a different test so the studentEmail should be correctly processed in the real version. Because this has been tested already and the goal of this test was to test the fetch function I thought it would be okay not to have to worry about the studentEmail and only about the fetch call.

Reply:
Ok, sounds good!